components/admin_console/group_settings/groups_list/groups_list.jsx

```diff
@@ -177,6 +197,178 @@ export default class GroupsList extends React.PureComponent {
         });
     }
 
+    regex = (str) => {
+        return new RegExp(`(${str})`, 'gi');
+    }
+
+    searchGroups = (page) => {
+        let {searchString} = this.state;
+
+        const newState = {...this.state};
+        delete newState.page;
+        delete newState.checked;
+
+        let q = searchString;
+        let opts = {q: ''};
+
+        Object.entries(FILTER_STATE_SEARCH_KEY_MAPPING).forEach(([key, value]) => {
+            const re = this.regex(value.filter);
+            if (re.test(searchString)) {
+                newState[key] = true;
+                q = q.replace(re, '');
+                opts = Object.assign(opts, value.option);
+            } else if (this.state[key]) {
+                searchString += ' ' + value.filter;
+            }
+        });
+
+        opts.q = q.trim();
+
+        newState.searchString = searchString;
+        newState.showFilters = false;
+        newState.loading = true;
+        newState.showFilters = false;
+        this.setState(newState);
+
+        this.props.actions.getLdapGroups(page, LDAP_GROUPS_PAGE_SIZE, opts).then(() => {
+            this.setState({loading: false});
+        });
+    }
+
+    handleGroupSearchKeyUp = (e) => {
+        const {key} = e;
+        const {searchString} = this.state;
+        if (key === Constants.KeyCodes.ENTER[0]) {
+            this.searchGroups();
+        }
+        const newState = {};
+        Object.entries(FILTER_STATE_SEARCH_KEY_MAPPING).forEach(([k, value]) => {
+            if (!this.regex(value.filter).test(searchString)) {
+                newState[k] = false;
+            }
+        });
+        this.setState(newState);
+    }
+
+    newSearchString = (searchString, stateKey, checked) => {
+        let newSearchString = searchString;
+        const {filter} = FILTER_STATE_SEARCH_KEY_MAPPING[stateKey];
+        const re = this.regex(filter);
+        const stringFilterPresent = re.test(searchString);
+
+        if (stringFilterPresent && !checked) {
+            newSearchString = searchString.replace(re, '').trim();
+        }
+
+        if (!stringFilterPresent && checked) {
+            newSearchString += ' ' + filter;
+        }
+
+        return newSearchString.replace(/\s{2,}/g, ' ');
+    }
+
+    handleFilterCheck = (updates) => {
+        let {searchString} = this.state;
+        updates.forEach((item) => {
+            searchString = this.newSearchString(searchString, item[0], item[1]);
+            this.setState({[item[0]]: item[1]});
+        });
+        this.setState({searchString});
+    }
+
+    renderSearchFilters = () => {
+        return (
+            <div
+                id='group-filters'
+                className='group-search-filters'
+                onClick={(e) => {
+                    e.nativeEvent.stopImmediatePropagation();
```

Comment:
@jespino React's the `stopPropagation` method of React's `SyntheticEvent` does not stop the propagation of native events (such as the one being used to handle clicks on `document`). One must access the native event via `SynteticEvent`'s `nativeEvent` property.